### PR TITLE
dev-federation-watcher examples

### PR DESCRIPTION
**Pull request by rodrigolc**

### CODE REVIEW

File: federation/src/watcher_example.c [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            spiffebundle_Bundle *bundle
+                = spiffebundle_Source_GetSpiffeBundleForTrustDomain(
+                    status->endpoint->source, status->endpoint->td, &err);
+            if(!bundle) {
+                fprintf(stderr, "Error retrieving bundle from source.\n");
+                printf("Can't find a bundle for \"%s\".\n",
+                       status->endpoint->td.name);
+                continue;
+            }
+            char *m_string = spiffebundle_Bundle_Marshal(bundle, &err);
```
Comment (otaviolcs3) [inline federation/src/watcher_example.c:36]:
```suggestion
            string_t m_string = spiffebundle_Bundle_Marshal(bundle, &err);
```

File: federation/src/watcher_example.c [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            if(m_string) {
+                printf("Bundle (jwks):\n%s\n", m_string);
+                util_string_t_Free(m_string);
+            } else {
+                printf("Bundle (jwks): ERROR\n");
+                fprintf(stderr, "Error marshalling bundle.\n");
+            }
+                }
```
Comment (otaviolcs3) [inline federation/src/watcher_example.c:44]: Apply code style

File: federation/src/watcher_example.c [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+#include "federation/src/watcher.h"
+#include "spiffeid/src/trustdomain.h"
+#include <check.h>
+#include <unistd.h>
+
+int running = true;
+
+int print_watcher(void *watcher_)
+{
+    spiffebundle_Watcher *watcher = watcher_;
```
Comment (otaviolcs3) [inline federation/src/watcher_example.c:11]:
```suggestion
    spiffebundle_Watcher *watcher = (spiffebundle_Watcher *) watcher_;
```

File: federation/src/watcher_example.c [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    // err = spiffebundle_Watcher_AddHttpsWebEndpoint(watcher, url2, td2);
+    // ...
+    // you can also add https_spiffe Endpoints:
+    // err = spiffebundle_Watcher_AddHttpsSpiffeEndpoint(watcher, url3, td3,
+    // id, source);
+    if(err != NO_ERROR) {
+        // free memory and exit
+        fprintf(stderr, "\nERROR ADDING ENDPOINT: %d\n", err);
+        spiffebundle_Watcher_Free(watcher);
+        return err;
```
Comment (otaviolcs3) [inline federation/src/watcher_example.c:73]: Better to use exit() function than return

File: federation/src/watcher_example.c [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    }
+    // starts watcher up, will not block
+    printf("Starting Watcher.\n");
+    err = spiffebundle_Watcher_Start(watcher);
+
+    if(err != NO_ERROR) {
+        // free memory and exit
+        fprintf(stderr, "\nERROR STARTING WATCHER: %d\n", err);
+        spiffebundle_Watcher_Free(watcher);
+        return err;
```
Comment (otaviolcs3) [inline federation/src/watcher_example.c:83]: Same here

File: federation/src/watcher_example.c [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    // start threads in the background
+    printf("Starting printing thread.\n");
+
+    thrd_t printer;
+    int thread_ret = thrd_create(&printer, print_watcher, watcher);
+    if(thread_ret != thrd_success) {
+        // free memory and exit
+        fprintf(stderr, "\nERROR STARTING THREAD: %d\n", err);
+        spiffebundle_Watcher_Free(watcher);
+        return err;
```
Comment (otaviolcs3) [inline federation/src/watcher_example.c:95]: Same here

File: federation/src/watcher_example.c [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+    running = false;
+
+    // stop watcher, blocks waiting for threads to close
+    err = spiffebundle_Watcher_Stop(watcher);
+    printf("Stopping Watcher.\n");
+    if(err != NO_ERROR) {
+        // free memory and exit
+        fprintf(stderr, "\nERROR STOPPING WATCHER: %d\n", err);
+        spiffebundle_Watcher_Free(watcher);
+        return err;
```
Comment (otaviolcs3) [inline federation/src/watcher_example.c:110]: Same

File: federation/src/watcher_example.c [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+                util_string_t_Free(m_string);
+            } else {
+                printf("Bundle (jwks): ERROR\n");
+                fprintf(stderr, "Error marshalling bundle.\n");
+            }
+                }
+    }
+}
+
+int main(int argc, char **argv)
```
Comment (otaviolcs3) [inline federation/src/watcher_example.c:48]:
```suggestion
int main(void)
```

**Review by arianalima — APPROVED** [review]

**Review by otaviolcs3 — APPROVED** [review]

**Pull request merged**